src/parser/monk/mistweaver/modules/spells/Celestial.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+
+import Analyzer from 'parser/core/Analyzer';
+import Combatants from 'parser/shared/modules/Combatants';
+
+import SpellIcon from 'common/SpellIcon';
+import StatisticBox, { STATISTIC_ORDER } from 'interface/others/StatisticBox';
+
+import { formatNumber } from 'common/format';
+import { TooltipElement } from 'common/Tooltip';
+
+const debug = false;
+
+class Celestial extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  petID = null;
+  soothHealing = 0;
+  soothOverHealing = 0;
+  envelopHealing = 0;
+  evelopOverhealing = 0;
+  healing = 0;
+  overhealing = 0;
+  count = 0;
+
+  on_byPlayer_summon(event) {
+    this.petID = event.targetID;
+    debug && console.log(`${event.ability.guid === SPELLS.INVOKE_CHIJI_THE_RED_CRANE_TALENT.id ? 'Chi-Ji' : 'Yu\'lon'} Summoned: ${this.petID}`);
+  }
+
+  on_heal(event) {
```

Comment:
on_heal will not do any (outside of the WCL query) filtering for the current active player or their pets; you check the event.sourceid against this.petid for soothing breath but don't do anything for enveloping breath. hope there is only one MW in the group! ;)

with old event listeners it's done with on_byPlayer_heal or on_byPlayerPet_heal (depending on the specific spell, if the event source is a pet or the player); with new ones its Events.heal.by(SELECTED_PLAYER) or SELECTED_PLAYER_PET

If the events are different (for example if enveloping is 'cast' by the player and soothing is 'cast' by the pet) you can split it up into two different functions or do the extra filtering inside the function itself to split up the logic